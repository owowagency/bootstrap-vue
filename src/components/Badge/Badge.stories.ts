import Badge from '.';
import Btn from '@/components/Button';
import {variants} from '@/composables/useVariant';

export default {
    title: 'Components/Badge',
    component: Badge,
    argTypes: {
        variant: {
            control: {type: 'select'},
            options: variants,
        },
    },
};

const template = (args) => ({
    components: {Badge},
    setup() {
        return {args};
    },
    template: '<Badge v-bind="args">I am a Badge</Badge>',
});

export const Default = template.bind({});

const variantsTemplate = (args) => ({
    components: {Badge},
    setup: () => ({args, variants}),
    template: '<Badge v-for="v of variants" class="me-1 mb-1" :variant="v" v-bind="args">{{ v }}</Badge>',
});

export const Variants = variantsTemplate.bind({});

export const Pill = variantsTemplate.bind({});

Pill.args = {
    pill: true,
};

export const Headings = (args) => ({
    components: {Badge},
    setup: () => ({args}),
    template: `
        <h1>Example heading <Badge>New</Badge></h1>
        <h2>Example heading <Badge>New</Badge></h2>
        <h3>Example heading <Badge>New</Badge></h3>
        <h4>Example heading <Badge>New</Badge></h4>
        <h5>Example heading <Badge>New</Badge></h5>
        <h6>Example heading <Badge>New</Badge></h6>
    `,
});

export const Buttons = (args) => ({
    components: {Badge, Btn},
    setup: () => ({args}),
    template: `
        <Btn class="me-1">Notifications <Badge variant="danger">4</Badge></Btn>
        <Btn class="position-relative">Notifications <Badge class="position-absolute top-0 start-100 translate-middle  border border-light" pill variant="danger">99+</Badge></Btn>
    `,
});
