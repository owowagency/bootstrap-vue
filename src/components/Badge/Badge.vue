<template>
    <div
        class="badge"
        :class="classes"
    >
        <!-- @slot Displays content inside the badge -->
        <slot />
    </div>
</template>

<script lang="ts">
import useVariant, {variantProps} from '@/composables/useVariant';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    /**
     * Indicates if the badge should be shown as a pill
     */
    pill: {
        type: Boolean,
        default: false,
    },
    ...variantProps,
});

const {classes} = useClasses(computed(() => [
    computed(() => props.pill ? 'rounded-pill' : undefined).value,
    useVariant(props.variant, 'bg-{0}').variantClass.value,
]));
</script>

<docs>
```vue
<Badge
    pill
    variant="danger"
>
    I am a badge
</Badge>
```

## Imported Props

| Prop name | Description                  | Type                                             | Values | Default   |
| --------- | ---------------------------- | ------------------------------------------------ | ------ | --------- |
| variant   | The variant of the component | [Variant](../../composables/useVariant) (string) | -      | 'primary' |
</docs>
