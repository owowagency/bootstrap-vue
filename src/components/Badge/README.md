# Badge

```vue
<Badge pill variant="danger">
    I am a badge
</Badge>
```

## Imported Props

| Prop name | Description                  | Type                                             | Values | Default   |
| --------- | ---------------------------- | ------------------------------------------------ | ------ | --------- |
| variant   | The variant of the component | [Variant](../../composables/useVariant) (string) | -      | 'primary' |

## Props

| Prop name | Description                                      | Type    | Values | Default |
| --------- | ------------------------------------------------ | ------- | ------ | ------- |
| pill      | Indicates if the badge should be shown as a pill | boolean | -      | false   |

## Slots

| Name    | Description                       | Bindings |
| ------- | --------------------------------- | -------- |
| default | Displays content inside the badge |          |
