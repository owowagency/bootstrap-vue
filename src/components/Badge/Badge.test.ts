import Badge from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(Badge);

    componentSlotRenderTest(Badge);

    componentWrapperClassTest(Badge, {pill: true}, 'rounded-pill');

    componentWrapperClassTest(Badge, {variant: 'danger'}, 'bg-danger');
});
