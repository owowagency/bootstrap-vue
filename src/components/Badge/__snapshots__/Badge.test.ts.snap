// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div class=\\"badge bg-primary\\">
  <!-- @slot Displays content inside the badge -->
</div>"
`;
